### ISSUE TITLE
crossref TDM auth

### ISSUE THREAD

**Issue by sckott** [sckott/textminer#13]
@selcukorkmaz  continuing from CrossRef/rest-api-doc/issues/96

We can continue discussion here on this topic. I emailed Elsevier, but still waiting to hear back on UC San Diego access

**Comment by selcukorkmaz** [comment]
@sckott I have tried the following using Elsevier API key:

```
curl -L -v "http://api.elsevier.com/content/article/PII:S0022283607007152?APIkey=
[Elsevier API key]&httpAccept=text/xml" > out.xml
```

That gives me abstract instead of full-text. When I check the Elsevier developer portal, I found this:

> By default, a request to http://api.elsevier.com/content/article/... will return:
> - the full-text XML version (FULL view) of the document, if your client IP address is recognized as that of a subscribing institute that has access to that article on ScienceDirect (which we call 'being entitled to the document');
> - the abstract XML version (META_ABS) of the document, if you're not entitled. 

This suggests that I am not entitled for the full-text paper. But as I said, I can access the full text article via Elsevier/ScienceDirect website, because I am using UC San Diego network.

**Comment by sckott** [comment]
Right, that's what happened to me too. There is another layer that needs to be fixed, a firewall that Elsevier can turn on or off for each institution. That's what I emailed them about, and once fixed, that should work for you

**Comment by selcukorkmaz** [comment]
Hi @sckott. The following is not working and I have no idea why.

```
curl -v -L -H "CR-Clickthrough-Client-Token: <CrossRef API Token>" 
"http://pubs.acs.org/doi/pdf/10.1021/jm1008727" > out.pdf
```

```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 198.29.7.70...
* Connected to pubs.acs.org (198.29.7.70) port 80 (#0)
> GET /doi/pdf/10.1021/jm1008727 HTTP/1.1
> Host: pubs.acs.org
> User-Agent: curl/7.43.0
> Accept: */*
> CR-Clickthrough-Client-Token: <CrossRef Token>
> 
< HTTP/1.1 302 Found
< Server: AtyponWS/7.1
< P3P: CP="NOI DSP ADM OUR IND OTC"
< Location: http://pubs.acs.org/doi/pdf/10.1021/jm1008727?cookieSet=1
< Set-Cookie: I2KBRCK=1; path=/; expires=Tue, 20-Dec-2016 19:03:55 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 95
< Date: Mon, 21 Dec 2015 19:03:55 GMT
< 
* Ignoring the response-body
{ [95 bytes data]
100    95  100    95    0     0    335      0 --:--:-- --:--:-- --:--:--   335100    95  100    95    0     0    335      0 --:--:-- --:--:-- --:--:--   334
* Connection #0 to host pubs.acs.org left intact
* Issue another request to this URL: 'http://pubs.acs.org/doi/pdf/10.1021/jm1008727?cookieSet=1'
* Found bundle for host pubs.acs.org: 0x7fbff06012d0
* Re-using existing connection! (#0) with host pubs.acs.org
* Connected to pubs.acs.org (198.29.7.70) port 80 (#0)
> GET /doi/pdf/10.1021/jm1008727?cookieSet=1 HTTP/1.1
> Host: pubs.acs.org
> User-Agent: curl/7.43.0
> Accept: */*
> CR-Clickthrough-Client-Token: <CrossRef Token>
> 
< HTTP/1.1 302 Found
< Server: AtyponWS/7.1
< Location: http://pubs.acs.org/action/cookieAbsent
< Content-Type: text/html; charset=utf-8
< Content-Length: 77
< Date: Mon, 21 Dec 2015 19:03:54 GMT
< 
* Ignoring the response-body
{ [77 bytes data]
100    77  100    77    0     0    212      0 --:--:-- --:--:-- --:--:--   212
* Connection #0 to host pubs.acs.org left intact
* Issue another request to this URL: 'http://pubs.acs.org/action/cookieAbsent'
* Found bundle for host pubs.acs.org: 0x7fbff06012d0
* Re-using existing connection! (#0) with host pubs.acs.org
* Connected to pubs.acs.org (198.29.7.70) port 80 (#0)
> GET /action/cookieAbsent HTTP/1.1
> Host: pubs.acs.org
> User-Agent: curl/7.43.0
> Accept: */*
> CR-Clickthrough-Client-Token: <CrossRef Token>
> 
< HTTP/1.1 200 OK
< Server: AtyponWS/7.1
< Cache-Control: no-cache
< Pragma: no-cache
< X-Webstats-RespID: 46303a9121ab8fa27b6c7f8bd341ca9d
< Set-Cookie: JSESSIONID=aaaJ6TP8Dbj0Hfcx6Iahv; path=/
< Content-Type: text/html; charset=UTF-8
< Date: Mon, 21 Dec 2015 19:03:53 GMT
< Connection: close
< 
{ [16097 bytes data]
100 65853    0 65853    0     0   118k      0 --:--:-- --:--:-- --:--:--  118k
* Closing connection 0
```

**Comment by sckott** [comment]
the click through token thing only works for publishers that use it. so far as far as I know, that's only Wiley and Elsevier, and one more small publisher (Journal of Zhejiang University Science)

for this article, I too can access in browser, but doesn't work on the command line

**Comment by selcukorkmaz** [comment]
I tried it without click through token but still it didn't give any result. I don't know if I need a publisher API key for this too.

**Comment by selcukorkmaz** [comment]
Well, this is just a big frustrating process. I can't access certain Elsevier articles, even though I have a subscription via my institution. Wiley only allow me download 60 papers at a time, which I need to download thousands of them. I can't even download other publishers' papers, which I have no idea why. 

I don't know why these people make this process harder and harder. This is not how science supposed to work.

**Comment by sckott** [comment]
Sorry about this. I don't have any control over this, But I have contacted more people at Elsevier to see if I can get your access fixed. Hopefully I'll hear back soon, but perhaps not give the holiday. Crossing fingers. 

you can try to get access through Elsevier directly using their API and api key, but maybe you have already tried that. 

For Wiley I was told the rate limit is `60 downloads per 10 minutes` - so you could adjust your requests to do 60 then wait until its been 10 or more minutes, then do another 60. I realize this is not ideal - hopefully with user feedback like yours and others, we can go to them requesting higher rate limits

**Comment by selcukorkmaz** [comment]
Thanks for your help @sckott!

**Comment by sckott** [comment]
@selcukorkmaz they flipped a switch, check again now trying to get Elsevier content either via their API or Crossref with click through token

**Comment by selcukorkmaz** [comment]
Thanks @sckott. It is working now. Thanks for your help. Appreciate it.

**Comment by sckott** [comment]
great! now that it works, i do maintain lowlevel Crossref clients in [Ruby](https://github.com/sckott/serrano), [Python](https://github.com/sckott/habanero), and [R](https://github.com/ropensci/rcrossref) if you use any of those langs. 

For textmining, I have a Ruby tool (this repo) which is useable for mining OA content from Crossref, and should work for Elsevier and Wiiley with your TDM token. Working on same capabilities in for R in https://github.com/ropensci/rcrossref and https://github.com/ropensci/fulltext

**Comment by selcukorkmaz** [comment]
@sckott I am using R. I will check R repos out.

**Comment by sckott** [comment]
great, let me know what you think
